src/main/resources/jupyter/edit-mode.js

```diff
@@ -0,0 +1,337 @@
+const dialog = require('base/js/dialog')
+const utils = require("base/js/utils")
+
+// TEMPLATED CODE
+// Leonardo has logic to find/replace templated values in the format $(...).
+let googleProject = $(googleProject);
+let clusterName = $(clusterName);
+
+define(() => {
+    let modalOpen = false
+    let meta = {}
+
+    const syncIssueButtons = {
+        'Make a Copy': {
+            click: () => saveAs(),
+            'class': 'btn-primary'
+        },
+        'Reload the workspace version and discard your changes': {
+            click: () => updateLocalCopyWithRemote()
+        }
+    }
+
+    const lockIssueButtons = {
+        'Run in Playground Mode': {
+            click: () => openPlaygroundMode(),
+            'class': 'btn-primary'
+        },
+        'Make a Copy': {
+            click: () => saveAs()
+        }
+    }
+
+    const noRemoteFileButtons = {
+        'Continue working': {
+            click: () => {},
+            'class': 'btn-primary'
+        }
+    }
+
+    const modeBannerId = "notification_mode"
+    const lockConflictTitle = "File is in use"
+    const syncIssueTitle = "File versions out of sync"
+    const syncIssueBody = "Your version of this file does not match the version in the workspace. What would you like to do?"
+    const syncIssueFatalBody = "This file was either deleted or never was stored with us."
+
+    //TODO URL resolution
+    const leoUrl = ''
+        // const leoUrl = 'http://localhost:8081'
+    const welderUrl = leoUrl + `/proxy/${googleProject}/${clusterName}/welder`
```

Comment:
Probably easier and can avoid the templating complexity: grab `Jupyter.notebook.base_url`, then replace "/notebooks/" with "/welder/"

Reply:
That doesn't match the Leonardo/Welder proxy URL pattern (see [cluster-site.conf](https://github.com/DataBiosphere/leonardo/blob/develop/src/main/resources/jupyter/cluster-site.conf).

Jupyter -> `leoBaseUrl/notebooks/${googleProject}/${clusterName}/jupyter`
Welder ->  `leoBaseUrl/proxy/${googleProject}/${clusterName}/welder/${pathToWelderEndpoint}`

We could make it more symmetrical as you suggest, but this would require adding a new proxy route to Leonardo, in addition to updating cluster-site.conf.  I chose the `/proxy/` pattern simply because that was easier: cluster-site.conf only.

Reply:
OK, this seems fine then.